chat-widget/src/components/livechatwidget/common/startChatErrorHandler.ts

```diff
@@ -0,0 +1,135 @@
+import { ChatSDKErrorName, ChatSDKError } from "@microsoft/omnichannel-chat-sdk";
+import { LogLevel, TelemetryEvent } from "../../../common/telemetry/TelemetryConstants";
+import { TelemetryHelper } from "../../../common/telemetry/TelemetryHelper";
+import { TelemetryTimers } from "../../../common/telemetry/TelemetryManager";
+import { ConversationState } from "../../../contexts/common/ConversationState";
+import { LiveChatWidgetActionType } from "../../../contexts/common/LiveChatWidgetActionType";
+import { Dispatch } from "react";
+import { ILiveChatWidgetAction } from "../../../contexts/common/ILiveChatWidgetAction";
+import { callingStateCleanUp, endChatStateCleanUp, closeChatStateCleanUp, chatSDKStateCleanUp } from "./endChat";
+import { NotificationHandler } from "../../webchatcontainerstateful/webchatcontroller/notification/NotificationHandler";
+import { NotificationScenarios } from "../../webchatcontainerstateful/webchatcontroller/enums/NotificationScenarios";
+import { DataStoreManager } from "../../../common/contextDataStore/DataStoreManager";
+import { ILiveChatWidgetProps } from "../interfaces/ILiveChatWidgetProps";
+import { getWidgetCacheIdfromProps } from "../../../common/utils";
+import { WidgetLoadCustomErrorString, WidgetLoadTelemetryMessage } from "../../../common/Constants";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const handleStartChatError = (dispatch: Dispatch<ILiveChatWidgetAction>, chatSDK: any, props: ILiveChatWidgetProps | undefined, ex: any, isStartChatSuccessful: boolean) => {
+    if (!ex) {
+        logWidgetLoadFailed();
+        return;
+    }
+
+    if (ex instanceof ChatSDKError) {
+        if (ex.message === ChatSDKErrorName.WidgetUseOutsideOperatingHour) {
+            dispatch({ type: LiveChatWidgetActionType.SET_OUTSIDE_OPERATING_HOURS, payload: true });
+            dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.OutOfOffice });
+            logWidgetLoadComplete(WidgetLoadTelemetryMessage.OOOHMessage);
+            return;
+        } else if (ex.message === ChatSDKErrorName.PersistentChatConversationRetrievalFailure ||
+            ex.message === ChatSDKErrorName.ConversationInitializationFailure ||
+            ex.message === ChatSDKErrorName.ChatTokenRetrievalFailure ||
+            ex.message === ChatSDKErrorName.UninitializedChatSDK) {
+            if (ex.httpResponseStatusCode === 400) {
+                logWidgetLoadFailed(ex);
+            } else {
+                logWidgetLoadCompleteWithError(ex);
+            }
+        } else if (ex.message === ChatSDKErrorName.InvalidConversation ||
```

Comment:
not a fan of if/else to handle the logic, maybe can we discuss use of enums or passing an object as error with error code and string, so we can have a switch block, and additional to that abstract the handling via strategy pattern for each scenario?, Im not a seer but I can predict that this method will grow to 100 LoC easily as we incorporate more cases, it will be better to break it by case and extract the functionality

Reply:
Changed the ChatSDK errors to use switch case